### ISSUE TITLE
[valid-expect-in-promise] has a bug for `then`/`catch`

### ISSUE THREAD

**Issue by shinjini-hopjump** [jest-community/eslint-plugin-jest#495]
Issue:
The following code triggers an error `Promise should be returned to test its fulfillment or rejection`:
```js
// error
new Promise().then(() => a.b())
```
Whereas the following does not:
```js
// no error
new Promise().then(() => a.b)
// no error
new Promise().then(() => a())
```
I also get the same results if I replace `then` with `catch`.

Expectation:
None of these should trigger errors since I am not expecting anything in them.

I have narrowed the issue to the `isFullExpectCall` function. Should you use `isExpectCall` from `utils` here? Using that causes the errors to go away.

I am on `23.1.1`

**Comment by G-Rath** [comment]
Sorry for the late reply.

I don't follow: the code snippets you've provided don't have any `expect` calls in them, nor do they trigger the rule by themselves.

Could you provide a more complete example, along with your eslint configuration?

**Comment by G-Rath** [comment]
Closing due to lack of information
